test/utils/PathUtils.test.ts

```diff
@@ -362,6 +370,63 @@ describe('PathUtils', () => {
         slices: ['Slice1', 'Slice2', 'Slice3']
       });
     });
+
+    it('should properly seperate path elements with slice names including brackets into PathParts', () => {
+      const testPath = 'item1[10][foo[x]].item2[11][bar[x]][baz[x]].value[x]';
+      const pathParts = parseFSHPath(testPath);
+
+      expect(loggerSpy.getAllLogs('error')).toBeEmpty();
+      expect(pathParts).toHaveLength(3);
+      expect(pathParts[0]).toEqual({
+        base: 'item1',
+        brackets: ['10', 'foo[x]'],
+        slices: ['foo[x]']
+      });
+      expect(pathParts[1]).toEqual({
+        base: 'item2',
+        brackets: ['11', 'bar[x]', 'baz[x]'],
+        slices: ['foo[x]', 'bar[x]', 'baz[x]']
+      });
+      expect(pathParts[2]).toEqual({
+        base: 'value[x]'
+      });
```

Comment:
I had forgotten that `parseFSHPath` doesn't include anything about previous slices for path parts that don't have any slices of their own. It's what we want, but I got scared for a minute.

Reply:
Yeah, me too. I originally rewrote this to include the previous slices even in this case -- and it broken a bunch of tests, so I reverted back to the original approach.